app/scripts/views/base.js

```diff
@@ -739,6 +745,7 @@ define(function (require, exports, module) {
         this.logError(nextViewData.error);
       }
 
+      this._hasNavigated = true;
```

Comment:
I'm picking some extremely tiny nits here, but if `trigger` threw for any reason (not sure if that's even possible), we'd have set `_hasNavigated` without navigation occurring. Is there a case to be made for setting the flag after `navigate` returns?